src/pages/product-page/product-modal/product-modal.tsx

```diff
@@ -0,0 +1,59 @@
+import styles from './product-modal.module.css';
+import ImageGallery from 'react-image-gallery';
+import {
+  SHOW_FULL_SCREEN_BUTTON,
+  SHOW_NAV,
+  SHOW_PLAY_BUTTON,
+  SHOW_THUMBNAILS,
+} from './constants.ts';
+import { useEffect } from 'react';
+
+type Image = {
+  original: string;
+};
+
+type Props = {
+  images: Image[];
+  onClick: () => void;
+  startIndex: number;
+};
+
+const defaultSettings = {
+  showThumbnails: SHOW_THUMBNAILS,
+  showFullscreenButton: SHOW_FULL_SCREEN_BUTTON,
+  showPlayButton: SHOW_PLAY_BUTTON,
+  showNav: SHOW_NAV,
+};
+
+const stopPropagation = (e: React.MouseEvent) => {
+  e.stopPropagation();
+};
+
+export const ProductModal = ({ images, startIndex, onClick }: Props) => {
+  useEffect(() => {
+    if (globalThis.window !== undefined && typeof document !== 'undefined') {
+      document.body.classList.add('noScroll');
+    }
+
+    return () => {
+      if (globalThis.window !== undefined && typeof document !== 'undefined') {
+        document.body.classList.remove('noScroll');
+      }
+    };
+  }, []);
```

Comment:
нет смысла в этих проверках. globalThis.window === undefined если код выполняется не в браузере, а на сервере. Такое происходит при SSR на нексте. в реакте нет SSR. твой код выполняется только в браузере. там всегда есть window. Более того, этот код написан в useEffect. а useEffect даже в нексте всегда выполняется только на клиенте

```suggestion
  useEffect(() => {
    document.body.classList.add('noScroll');
  
    return () => {
      document.body.classList.remove('noScroll');
    };
  }, []);
```